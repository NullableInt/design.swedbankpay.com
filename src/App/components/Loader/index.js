import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const Loader = ({ id, size, muted, visible }) => {
    const loaderClasses = classnames(
        "loader",
        size ? `loader-${size}` : null,
        muted ? "loader-muted" : null,
        visible ? "in" : null
    );

    return <div id={id} className={loaderClasses} data-loader="true"></div>;
};

Loader.propTypes = {
    id: PropTypes.string,
    size: PropTypes.oneOf(["small", "default", "large"]).isRequired,
    muted: PropTypes.bool,
    visible: PropTypes.bool
};

export default Loader;
