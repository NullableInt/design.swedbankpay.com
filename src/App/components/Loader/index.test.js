import React from "react";
import { shallow } from "enzyme";

import Loader from "./index";

describe("Component: Loader - ", () => {
    it("is defined", () => {
        expect(Loader).toBeDefined();
    });

    it("prop size is marked as required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<Loader />);

        expect(wrapper).toMatchSnapshot();
        expect(console.error).toHaveBeenCalled();
    });

    it("renders a default loader with the specified size when prop size is provided", () => {
        const wrapper = shallow(<Loader size="test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("loader loader-test");
    });

    it("renders a muted loader with the specified size when prop size and muted is provided", () => {
        const wrapper = shallow(<Loader size="test" muted />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("loader loader-test loader-muted");
    });
});
