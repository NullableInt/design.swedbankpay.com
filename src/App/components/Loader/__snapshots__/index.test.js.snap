// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader -  prop size is marked as required 1`] = `
<div
  className="loader"
  data-loader="true"
/>
`;

exports[`Component: Loader -  renders a default loader with the specified size when prop size is provided 1`] = `
<div
  className="loader loader-test"
  data-loader="true"
/>
`;

exports[`Component: Loader -  renders a loader with the specified size and class in when prop size and visible is provided 1`] = `
<div
  className="loader loader-test in"
  data-loader="true"
/>
`;

exports[`Component: Loader -  renders a muted loader with the specified size when prop size and muted is provided 1`] = `
<div
  className="loader loader-test loader-muted"
  data-loader="true"
/>
`;
